### PR TITLE
{Keyvault} Support managed HSM private link operations

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

Support MHSM private link operations in keyvault module and network module:
- list private link resource
- approve/reject/delete/show private endpoint connection

**Testing Guide**

See scenario tests

**History Notes**

[Keyvault] `az keyvault private-link-resource list`: Support MHSM
[Keyvault] `az keyvault private-endpoint-connection`: Support MHSM
[Network] `az network private-link-resource list`: Support `--type microsoft.keyvault/managedHSMs`
[Network] `az network private-endpoint-connection`: Support `--type microsoft.keyvault/managedHSMs`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Keyvault

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 1]
```diff
@@ -145,6 +157,104 @@ def test_private_endpoint_connection_keyvault(self, resource_group):
...
+                 checks=self.check('privateEndpointNetworkPolicies', 'Disabled'))
+
+        # Create a private endpoint connection
+        pe = self.cmd('network private-endpoint create '
+                      '-g {rg} '
+                      '-n {pe} '
+                      '--vnet-name {vnet} '
+                      '--subnet {subnet} '
+                      '-l {loc} '
+                      '--connection-name {pe_connection} '
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:196]: The variable name `pe` is not accurate for `--connection-name`.

```
> az network private-endpoint create -h

--connection-name                [Required] : Name of the private link service connection.
```

Should be something like `pl`.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:196]: It's `private_endpoint`'s `private_link_service_connection`.
https://github.com/Azure/azure-cli/blob/e0d7b53da27cd6c7242456cfd8cecc00fe5fe04a/src/azure-cli/azure/cli/command_modules/network/custom.py#L3063

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py [hunk 2]
```diff
@@ -105,6 +117,22 @@ def test_keyvault_private_link_resource(self, resource_group):
...
 
+class KeyVaultMHSMPrivateLinkResourceScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_hsm_plr_rg')
+    def test_mhsm_private_link_resource(self, resource_group):
+        self.kwargs.update({
+            'hsm': self.create_random_name('cli-test-hsm-plr-', 24),
+            'loc': 'centraluseuap'
+        })
+        _create_hsm(self)
+        self.cmd('keyvault private-link-resource list --hsm-name {hsm}',
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:128]: What is the difference between `az network private-link-resource list`?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:128]: From `--debug` log I can see `az network private-link-resource list` uses `send_raw_request` (`az rest`) but `az keyvault private-link-resource list` uses Track 2 SDK. This lead to duplication and confusion.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:128]: Yes it's using the same request. Keyvault send the request with SDK and Network send with raw request.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 3]
```diff
@@ -145,6 +157,104 @@ def test_private_endpoint_connection_keyvault(self, resource_group):
...
+                 '--type Microsoft.KeyVault/managedHSMs',
+                 checks=self.check('name', '{hsm_pe_name}'))
+
+        # Test approval/rejection
+        self.kwargs.update({
+            'approval_desc': 'You are approved!',
+            'rejection_desc': 'You are rejected!'
+        })
+
+        self.cmd('network private-endpoint-connection approve '
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:223]: Similar test cases are repeated 4 times, becoming a Cartesian product of {vault, hsm} × {network, keyvault}. This definitely leads to maintenance difficulty.

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:223]: True. We could refactor it later

**Review by jiasli — APPROVED** [review]

**Pull request merged**